### PR TITLE
feat: added cart management

### PR DESCRIPTION
**Pull request by xii-vi**
#### Remove from Cart

mobile (13)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -4,16 +4,22 @@ import { Routes, Route } from "react-router-dom";
 import { ProductListing } from './pages/product/productListing';
 import { Login } from './pages/login/login';
 import { Signup } from './pages/signUp/signup';
+import { Cart } from './pages/cart/cart';
+import { Navbar } from './component/navbar/navbar';
+import { Footer } from './component/footer/footer';
```
Comment (devdeadviz) [inline src/App.js:9]: You can add an `index.js` file in your pages folder and add all your page exports in that file, then you will be able to import all your pages from a single file.

File: src/apis/apis.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        localStorage.setItem('userData', JSON.stringify(updatedCartData))
+        }
+        catch (error) {
+        console.log(error);
+        }
+
+}
+
+const IncreaseCartQuantity = async (productData, encodedToken) => {
+console.log(productData)
```
Comment (devdeadviz) [inline src/apis/apis.js:40]: Remove `console.logs` statements from your codebase.

File: src/component/card/card.jsx [hunk 3]
```diff
@@ -1,7 +1,28 @@
-import { Link } from "react-router-dom";
+import { Link , useNavigate} from "react-router-dom";
+import { useCart } from "../../context/cartContext";
+import { useAuth } from "../../context/authContext";
```
Comment (devdeadviz) [inline src/component/card/card.jsx:3]: You can add an `index.js` file in your context folder and add all your context exports in that file, then you will be able to import all your contexts from a single file.

File: src/component/card/cartProductCard.jsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+export const CartProductCard = (props)=>{
+  const {
+        authState: { token },
+      } = useAuth();
+  
+    const {
+      cartDispatch,
+    } = useCart();
+
+  const data = props.singleP
```
Comment (devdeadviz) [inline src/component/card/cartProductCard.jsx:17]: What is singleP? Try to be more descriptive.

File: src/component/card/cartProductCard.jsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+                    alt={data.model} />
+                <div className="mx-5 pt-5 pb-2">
+                    <small>{data.brand}</small>
+                    <p className="py-2">{data.model}</p>
+                    <small>₹ {data.price}</small>
+                    <div className="my-4">
+                      {data.quantity === 1 ? (<button className="p-1" onClick={decreaseCartHandler}>
+                        <i className="fas fa-trash"></i>
+                    </button>):(<button onClick={decreaseCartHandler}><i className="fa fa-minus p-1"></i></button> ) }
+                        <span className="text-bold mx-2">{data.quantity}</span>
```
Comment (devdeadviz) [inline src/component/card/cartProductCard.jsx:50]: You can further destructure your data so as avoid writing `data.something`.

File: src/reducer/cartReducer.jsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
+export const addToCartReducer=(state, dispatch)=>{
```
Comment (devdeadviz) [inline src/reducer/cartReducer.jsx:1]: Just a suggestion, generally we use `action` keyword instead of `dispatch` in the reducer function.

**Review by devdeadviz — APPROVED** [review]
The code is clean and your file structure is also good, you just need to take care of some corner cases. Good work!

**Review by ivinayakg — APPROVED** [review]
the code looks good, just one suggestion you are heavily focusing on localstorage for state management try to di from the context itself will work better.
Great Work 👍

**Pull request merged**